torch/csrc/jit/passes/create_autodiff_subgraphs.cpp

```diff
@@ -0,0 +1,109 @@
+#include <cstddef>
+#include "torch/csrc/jit/ir.h"
+#include "torch/csrc/jit/autodiff.h"
+
+namespace torch { namespace jit {
+
+struct Graph;
+
+namespace {
+
+// Move nodes that exist in graph g into a 'group_node_kind' node.
+// All inputs shared by the nodes become inputs to the new node.
+// Outputs from 'nodes' are redirected to outputs of the new node,
+// and the original nodes are removed.
+// prereq: it is topologically valid to place the new node
+// right before nodes[0] (i.e. it will not create cycles and all uses of
+// new node will be after this position).
```

Comment:
This is a weaker requirement than requiring the nodes be contiguous. However, is this requirement equivalent to saying that you could float declarations up/down topsort such that they are contiguous?